### PR TITLE
Rename X509_OBJECT_up_ref_count->X509_OBJECT_up_ref

### PR DESCRIPTION
**Pull request by richsalz**
This is the just about the smallest of nits, but I noticed it while writing the manpage in #13788. It's really an internal undocumented function that nobody should ever need to call, but consistency counts.

### CODE REVIEW

**Comment by t8m** [comment]
This looks good except for one thing - we have now this rule of not adding any new undocumented function and by this rename you're effectively adding it.

Maybe we should have a generic up_ref manpage where all the up_ref calls should be added and perhaps some of the text from the new threading manpage moved to?

**Comment by richsalz** [comment]
I understand the rule about not adding new undocumented functions, and I agree with it, even though it's likely to end of torpedoing another cleanup PR because someone who added functions did not document them; see the thread at https://github.com/openssl/openssl/pull/13742#discussion_r551387954.

But more importantly, many (including myself and my employer) are frustrated with the delays in 3.0.  I have been trying to help by adding thing in the 3.0 milestone. I had convinced myself that #13788 was related because of the large number of questions that come in about thread-safety.  I first thought that this was a simple cleanup that I noticed while researching for the thread-safety manpage. Then I got concerned someone would worry about deprecation, so I handled that.  And now, like #13742, this bumps into the documentation requirement.

I really don't want the project to be distracted by things that are not in the [3.0 milestone](https://github.com/openssl/openssl/milestone/17) list.  (Man, I do a "headdesk" whenever I see yet another x509/apps cleanup PR)

TL;DR This isn't a 3.0 thing, I'm closing it.

**Comment by petrovr** [comment]
Rich Salz wrote:
> This is the just about the smallest of nits, but I noticed it while writing the manpage in #13788. It&#39;s really an internal undocumented function that nobody should ever need to call, but consistency counts.
>
> You can view, comment on, or merge this pull request online at:
>
>    https://github.com/openssl/openssl/pull/13816
>
> -- Commit Summary --
>
>    * Rename X509_OBJECT_up_ref_count-&gt;X509_OBJECT_up_ref
This is wrong . There is no counter on object (argument).

Other methods increment argument while this function increments "related data"

Regards,
Roumen Petrov

**Comment by richsalz** [comment]
I think you're wrong, @petrovr .  See https://github.com/openssl/openssl/blob/master/crypto/x509/x509_lu.c#L407 and the functions that it calls.
